client/components/payment-method-missing-currency-pill/index.js

```diff
@@ -0,0 +1,54 @@
+import { __, sprintf } from '@wordpress/i18n';
+import React, { useContext } from 'react';
+import styled from '@emotion/styled';
+import PaymentMethodsMap from '../../payment-methods-map';
+import Pill from 'wcstripe/components/pill';
+import Tooltip from 'wcstripe/components/tooltip';
+import UpeToggleContext from 'wcstripe/settings/upe-toggle/context';
+
+const StyledPill = styled( Pill )`
+	border: 1px solid #f0b849;
+	background-color: #f0b849;
+	color: #1e1e1e;
+	line-height: 16px;
+`;
+
+const PaymentMethodMissingCurrencyPill = ( { id, label } ) => {
+	const { isUpeEnabled } = useContext( UpeToggleContext );
+	const paymentMethodCurrencies = PaymentMethodsMap[ id ]?.currencies || [];
+	const storeCurrency = window?.wcSettings?.currency?.code;
+
+	if ( ! isUpeEnabled ) {
+		return null;
+	}
+
+	const neededCurrency = paymentMethodCurrencies.find(
+		( currency ) => currency !== storeCurrency
+	);
+
+	if (
+		id !== 'card' &&
+		! paymentMethodCurrencies.includes( storeCurrency )
+	) {
+		return (
+			<Tooltip
+				content={ sprintf(
+					/* translators: $1: a payment method name. %2: Currency. */
+					__(
+						"%1$s won't be visible to your customers until you add %2$s to your store."
+					),
+					label,
+					neededCurrency
+				) }
```

Comment:
I agree we probably won't get to the edge case where WCPay is enabled along with Stripe, but there are other multi-currency plugins, e.g [1](https://woocommerce.com/products/multi-currency-switcher/), [2](https://woocommerce.com/products/currency-switcher-for-woocommerce/), [3](https://woocommerce.com/products/currency-converter-widget/), [4](https://woocommerce.com/products/multi-currency/).

Of course we don't need to check for all these extensions and hide the pill, but I think we can improve the copy so that it doesn't exclude the possibility of a multi-currency plugin being used.

More specifically, the direction "until you add %2$s to your store." kind of implicitly says we're not aware of any multi-currency extensions and the user needs to do something.

I guess we could use the old settings copy:

<img width="224" alt="Old settings missing currency" src="https://user-images.githubusercontent.com/5509901/141026107-902b8f74-f759-4da4-aa97-9ad226725c76.png">

And instead of filtering one of the missing currencies, we should probably mention them all:

```JavaScript
const neededCurrencies = paymentMethodCurrencies.join(', ');
```

Feel free to suggest something else for the tooltip as well. Perhaps the old settings one is not ideal for when you have just one missing currency, which is the most common scenario.

Reply:
@cesarcosta99 Also, do we want to add the missing currency pill to non-UPE payment methods as well? Like the pending activation pill.

<img width="951" alt="Pending activation pill non-UPE APMs" src="https://user-images.githubusercontent.com/5509901/141031894-32a113ce-7fcc-463f-a00a-03a50eddf860.png">

Reply:
Thanks for taking the time to review this!

>so that it doesn't exclude the possibility of a multi-currency plugin being used.

Good point, however UPE uses the default store currency. That means if I consider whatever multi-currency plugin and hide the notice because they contain the currency the payment method requires, that will wrongly indicate that that PM will work on UPE, which turns out to be false.

>And instead of filtering one of the missing currencies, we should probably mention them all

True enough. Addressed in 49fc386.

>@cesarcosta99 Also, do we want to add the missing currency pill to non-UPE payment methods as well? Like the pending activation pill.

Yes, will create an issue to address it in another moment as we talked.